kotlin/change-basemaps/src/main/java/com/esri/arcgisruntime/sample/changebasemaps/MainActivity.kt

```diff
@@ -0,0 +1,164 @@
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import android.view.MenuItem
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private lateinit var mNavigationDrawerItemTitles: Array<String>
+
+  private val mDrawerToggle: ActionBarDrawerToggle by lazy { setupDrawer() }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // inflate navigation drawer with all basemap types in a human readable format
+    mNavigationDrawerItemTitles =
+        Basemap.Type.values().map { it.name.replace("_", " ").toLowerCase().capitalize() }.toTypedArray()
+
+    addDrawerItems()
+    drawerLayout.addDrawerListener(mDrawerToggle)
+
+    supportActionBar?.apply {
+      setDisplayHomeAsUpEnabled(true)
+      setHomeButtonEnabled(true)
+      // set opening basemap title to Topographic
+      title = mNavigationDrawerItemTitles[2]
+    }
+
+    // create a map with Topographic Basemap and set it to the map
+    mapView.map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 47.6047381, -122.3334255, 12)
+  }
+
+  /**
+   * Add navigation drawer items
+   */
+  private fun addDrawerItems() {
+    ArrayAdapter(this, android.R.layout.simple_list_item_1, mNavigationDrawerItemTitles).apply {
+      drawerList.adapter = this
+      drawerList.onItemClickListener =
+          AdapterView.OnItemClickListener { adapterView, view, position, id -> selectBasemap(position) }
+    }
+  }
+
+  /**
+   * Set up the navigation drawer
+   */
+  private fun setupDrawer() =
+      object : ActionBarDrawerToggle(this, drawerLayout, R.string.drawer_open, R.string.drawer_close) {
+
+        override fun isDrawerIndicatorEnabled() = true
+
+        /** Called when a drawer has settled in a completely open state.  */
+        override fun onDrawerOpened(drawerView: View) {
+          super.onDrawerOpened(drawerView)
+          supportActionBar?.title = title
+          invalidateOptionsMenu() // creates call to onPrepareOptionsMenu()
+        }
+
+        /** Called when a drawer has settled in a completely closed state.  */
+        override fun onDrawerClosed(view: View) {
+          super.onDrawerClosed(view)
+          invalidateOptionsMenu() // creates call to onPrepareOptionsMenu()
+        }
+      }
+
+  /**
+   * Select the Basemap item based on position in the navigation drawer
+   *
+   * @param position order int in navigation drawer
+   */
+  private fun selectBasemap(position: Int) {
+    // update selected item and title, then close the drawer
+    drawerList.setItemChecked(position, true)
+    drawerLayout.closeDrawer(drawerList)
+
+    Log.d("before", mNavigationDrawerItemTitles[position])
```

Comment:
are these `log` statements supposed to be in the final sample?

Reply:
Oops, thanks! No they're not!